src/ape/__init__.py

```diff
@@ -31,39 +32,47 @@
     },
     # What we are considering to be the starting project directory
     PROJECT_FOLDER=_Path.cwd(),
-    plugin_manager=plugin_manager,
 )
 """The active configs for the current project. See :class:`ape.managers.config.ConfigManager`."""
 
 # Main types we export for the user
-
-compilers = _CompilerManager(config, plugin_manager)  # type: ignore
+_CompilerManager.config = config
+_CompilerManager.plugin_manager = plugin_manager
+compilers = _CompilerManager()
 """Manages compilers for the current project. See
 :class:`ape.managers.compilers.CompilerManager`."""
 
-networks = _NetworkManager(config, plugin_manager)  # type: ignore
+_NetworkManager.config = config
+_NetworkManager.plugin_manager = plugin_manager
+networks = _NetworkManager()
 """Manages the networks for the current project. See
 :class:`ape.managers.networks.NetworkManager`."""
 
-_converters = _ConversionManager(config, plugin_manager, networks)  # type: ignore
+_ConversionManager.config = config
+_ConversionManager.plugin_manager = plugin_manager
+_ConversionManager.networks = networks
+_converters = _ConversionManager()
 
-chain = _ChainManager(networks)  # type: ignore
+_ChainManager._networks = networks
+chain = _ChainManager()
 """
 The current connected blockchain; requires an active provider.
 Useful for development purposes, such as controlling the state of the blockchain.
 Also handy for querying data about the chain and managing local caches.
 """
 
-accounts = _AccountManager(config, _converters, plugin_manager, networks)  # type: ignore
+_AccountManager.config = config
+_AccountManager.converters = _converters
+_AccountManager.plugin_manager = plugin_manager
+_AccountManager.network_manager = networks
+accounts = _AccountManager()
 """Manages accounts for the current project. See :class:`ape.managers.accounts.AccountManager`."""
 
-Project = _partial(
-    _ProjectManager,
-    config=config,
-    compilers=compilers,
-    networks=networks,
-    converter=_converters,
-)
+_ProjectManager.config = config
+_ProjectManager.compilers = compilers
+_ProjectManager.networks = networks
+_ProjectManager.converter = _converters
+Project = _partial(_ProjectManager)
```

Comment:
at this point, just import `ProjectManager as Project`

Reply:
I also like how this removes partial

Reply:
p.s. `Contract` down below (which also is a partial` should probably be a full class, and then we can remove the partial as well through DI